Makefile

```diff
@@ -160,7 +160,7 @@ coverage: $(INSTALL_DIR)/koreader/.luacov
 fetchthirdparty:
 	git submodule init
 	git submodule sync
-	git submodule update
+	git submodule update --depth 1 --jobs 3
```

Comment:
GitHub allows direct SHA fetch now? A few years ago this would've been guaranteed to fail eventually, but `l18n/` not causing any problems would seem to suggest that's no longer an issue.

Reply:
Additionally, if we can indeed reliably do it this way now for CI that's fantastic, but before we merge we might want to consider the developer use case as well, which desires the current behavior (unless that's just me). At its simplest something like an additional `fetchthirdparty-full` but there might be a more elegant solution.

Reply:
How about [this instead](https://github.com/koreader/koreader/compare/6a13735a82164f58585167cc63be278f2dfde7ed..80d102359a96f9ef7703d365727767d4f2ff6dbe)? And I'll make a corresponding PR for `fetchthirdparty` in base.

Reply:
`--recommend-shallow` is the default, isn't it? Simply making `l10n` as shallow should suffice. But yes, that seems fine.

Reply:
It does not seem to be the default, no. Both on Arch Linux with 2.44.0 and our docker base image using git 2.25.1, I get a full fetch of the l10n submodule when not explicitly using `--recommend-shallow`.

Reply:
Or rather I can't seem to get it to work with the `l10n` submodule…

Reply:
The saved hash is quite far removed from master even though it's only a few weeks old, basically the thing that would've been guaranteed to fail a few years ago. Which is to say, shallow could still mean hundreds of commits?

Reply:
Yeah, that seem to be why: if I bump l10n to master, clone that repo, and `git submodule update --init l10n`, I get the correct behavior.

Reply:
(With the additional `shallow = true` entry in `.gitmodules`)

Reply:
And with `--recommend-shallow` it also only downloads a single commit? That seems unexpected, but in that case it can stay.

Reply:
Nope, not for that `l10n` one. You were right: `--recommend-shallow` is the default, but the behavior is not necessarily the same as passing `--depth 1`.

Reply:
There's a pretty big difference between (tens of?) thousands and a few hundred either way, so I wouldn't consider it a loss.

Also note that on release builds the script currently pulls in the latest master, for better and worse. (I don't think there's been a worse yet but you never know. :-) So it'll happen at some point regardless in that case.

Reply:
Locally, it's the difference between a ~15s `make fetchthirdparty` with a ~10MB `.git/modules/l10n` when forcing a shallow clone of `l10n` and 1m30 / 1G with master.

The behavior also seems fine when using `make po` after a forced shallow clone of `l10n`. (The shallow clone does initially end up with 2 commits, tip of master and the current submodule ref).

Reply:
Hm yeah that wouldn't be shallow, that's simply the whole thing.